core/unix/memquery_test.h

```diff
@@ -0,0 +1,311 @@
+/* *******************************************************************************
+ * Copyright (c) 2019 Google, Inc.  All rights reserved.
+ * *******************************************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/*
+ * memquery_test.h - unit test for unix memquery logic, it is meant to be
+ * included into memquery.c after its includes.
+ *
+ * See also memquery_test_data.h
+ */
+
+#define MEMQUERY_ITERATOR_START memquery_iterator_start_test
+#define MEMQUERY_ITERATOR_NEXT memquery_iterator_next_test
+#define MEMQUERY_ITERATOR_STOP memquery_iterator_stop_test
+#define MODULE_IS_HEADER module_is_header_test
+#define MODULE_WALK_PROGRAM_HEADERS module_walk_program_headers_test
+
+typedef struct {
+    memquery_iter_t iter_result;
+    bool is_header;
+    app_pc mod_base, mod_end;
+} fake_memquery_result;
+
+typedef struct {
+    fake_memquery_result *iters;
+    size_t iters_count;
+
+    const char *in_name;
+    app_pc in_start;
+
+    int want_return;
+    app_pc want_start;
+    app_pc want_end;
+    const char *want_fulldir;
+    const char *want_filename;
+} memquery_library_bounds_test;
+
+#include "memquery_test_data.h"
```

Comment:
Oh I definitely clang-formatted this, I didn't do it by hand! (and I didn't want to waste time getting the record_ function to output correctly formatted code.) I'm not sure why there was this line here, perhaps it was an accidental update while I was hand-picking test cases. Removed.

Reply:
Re the numbers: I have a post-processing script which automatically parses output from the test suite and replaces the TODORENAME with an increasing number; after that point I manually removed the test cases which seemed repetitious.

I may re-generate all of these test cases due to some other requests you have; stay tuned.

Reply:
It took me some time to come up with a solution on how to automatically name the tests since I want them to be decoupled from any existing tests by default. I ended up using the current timestamp (in hex) as each the unique identifier for both the test and fake maps result. It looks /slightly/ less jarring that the gaps in these test numbers at least to me.